Tchap/Modules/Room/RoomViewController.m

```diff
@@ -2581,6 +2581,60 @@ - (void)showAdditionalActionsMenuForEvent:(MXEvent*)selectedEvent inCell:(id<MXK
         }
 #endif
         
+        NSString *titleKey;
+        if (![self.roomDataSource.room.accountData getTaggedEventInfo:selectedEvent.eventId withTag:kMXRoomTagFavourite])
+        {
+            titleKey = @"room_event_action_favourite";
+        }
+        else
+        {
+            titleKey = @"room_event_action_remove_favourite";
+        }
+        [currentAlert addAction:[UIAlertAction actionWithTitle: NSLocalizedStringFromTable(titleKey, @"Vector", nil)
+          style:UIAlertActionStyleDefault
+        handler:^(UIAlertAction * action) {
+            
+            if (weakSelf)
```

Comment:
FYI you may use "MXStrongifyAndReturnIfNil(self);" to  simplify the code (I forgot to mention it)
I let you check its use combined with "MXWeakify(self);"

Reply:
I updated it

Reply:
please see my new comment about your changes